assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,238 @@
+} ) => {
+	// eslint-disable-next-line camelcase
+	const mappedChoices = choices.map( ( { answer_ordinal, write_in } ) => {
+		// write_in means we need to support free text for that option (answer_text holds this in state).
+		const optionalKeys = write_in ? { answer_text: '' } : {}; // eslint-disable-line camelcase
+
+		return {
+			answer_ordinal,
+			selected: false,
+			...optionalKeys,
+		};
+	} );
+
+	const initialState = keyBy( mappedChoices, 'answer_ordinal' );
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	// eslint-disable-next-line camelcase
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	// eslint-disable-next-line camelcase
+	const handleAnswerChange = ( event, answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				answer_text: event.target.value?.slice(
+					0,
+					MAXIMUM_CHARACTER_LIMIT
+				),
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleSubmit = () => {
+		const answer = Object.values( selectedValues )
+			.filter( ( { selected } ) => selected )
+			// eslint-disable-next-line camelcase
+			.map( ( { answer_ordinal, answer_text } ) => {
+				// eslint-disable-next-line camelcase
+				if ( answer_text ) {
+					return { answer_ordinal, answer_text };
+				}
+				return { answer_ordinal };
+			} );
+		answerQuestion( {
+			answer,
+		} );
+	};
+
+	const hasEmptySelectedTextValue =
+		// eslint-disable-next-line camelcase
+		choices.filter( ( { write_in, answer_ordinal } ) => {
+			// eslint-disable-next-line camelcase
+			if ( write_in ) {
+				// eslint-disable-next-line camelcase
+				const { selected, answer_text } = selectedValues[
+					answer_ordinal
+				];
+				if ( selected && answer_text.length === 0 ) {
+					return true;
+				}
+			}
+			return false;
+		} ).length > 0;
+
+	const totalSelectedValues = Object.values( selectedValues ).filter(
+		( { selected } ) => selected
+	).length;
+	const hasLessThanMinChoices = totalSelectedValues < minChoices;
+	const hasMoreThanMaxChoices =
+		maxChoices && totalSelectedValues > maxChoices;
+
+	const isSubmitButtonDisabled =
+		hasEmptySelectedTextValue ||
+		hasLessThanMinChoices ||
+		hasMoreThanMaxChoices;
+
+	return (
+		<div className="googlesitekit-survey__multi-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+
+			<div className="googlesitekit-survey__body">
+				{
+					// eslint-disable-next-line camelcase
+					choices.map( ( { answer_ordinal, text, write_in } ) => {
+						const answer = selectedValues[ answer_ordinal ];
+						// eslint-disable-next-line camelcase
+						const uniqueID = `googlesitekit-survey__multi-select-${ answer_ordinal }-${ text }`;
```

Comment:
We can create unique instance IDs using hooks instead, we do it elsewhere: https://github.com/google/site-kit-wp/blob/ee5e21191618450ad6a10332400905e81f7a16cb/assets/js/components/Dialog.js#L60 🤷🏻‍♂️

Reply:
I can't use that here though as it's in a loop.  So not only would the Ids all be the same, but `eslint-plugin-react` won't allow this (and I'm not sure what would actually happen in React! :sweat_smile: )

I could create one `instanceID` and then append `answer_ordinal` to it

Alternatives:
1. Create a new inner Choice component (like on the SingleSelect PR)
1. Wrap the `TextField` inside the `label` element.

Reply:
See other comment, I went with 2.

Reply:
Ah right, whoops, didn't put together the loop piece.

We don't tend to wrap components with labels but I've not problem with that 👍🏻